ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp

```diff
@@ -267,15 +268,50 @@ bool window_get_stayontop() {
 }
 
 void window_set_sizeable(bool sizeable) {
-  if (enigma::isSizeable == sizeable) return;
```

Comment:
This here is the questionable part of why I am not sure to approve it without asking Josh. Rather than eliminate this redundancy check, maybe it would be better to gut your code below into an internal engine function `enigma::window_update_size_hints` above this line in the source. Then update all of the setters to call the internal engine function. Generally, it doesn't make sense to proceed the sizeable change if the window is already sizeable etc.

Reply:
Another questionable thing about this is that you are always setting `PMinSize` and `PMaxSize` even when only one of them is changed. That may not be slow or have any side effects, but the window manager does have these separate for a reason. Maybe check if the parameters are `-1` or different from the previous state and only set the hint that's actually changing. We do this in the display functions too iirc.

Reply:
@JoshDreamland

Reply:
As discussed on Discord, this should be fine the way it is. I don't want to forbid an update because our window state cache went stale. Linux desktop environments tend to do whatever they want with your window, at the user's request. You full-screen it, they'll un-full-screen it. You pin it, they'll unpin it. Wouldn't surprise me if setting a window to be resizable was a first-class feature of some DE.